mcsema/BC/Function.cpp

```diff
@@ -260,6 +303,123 @@ static llvm::BasicBlock *GetOrCreateBlock(TranslationContext &ctx,
+      auto lpad = ir.CreateLandingPad(exn_type, personality, 1, "cleanup.lpad");
+#endif
+      lpad->setCleanup(true);
+
+      auto exct_value_0  = ir.CreateExtractValue(lpad, 0);
+      auto store_loc_0 = ir.CreateAlloca(exct_value_0->getType());
+      ir.CreateStore(exct_value_0, store_loc_0);
+
+      auto exct_value_1 = ir.CreateExtractValue(lpad, 1);
+      auto store_loc_1 = ir.CreateAlloca(exct_value_1->getType());
+      ir.CreateStore(exct_value_1, store_loc_1);
+
+      // if `ctx.ea_to_block.count(entry->lp_ea) == 0`, the landing pad basic block
+      // might not be recovered. Throw a warning in such case.
+      LOG_IF(ERROR, !ctx.ea_to_block.count(entry->lp_ea))
+          << "Missing block at " << std::hex << entry->lp_ea
+          << " for exception landing pad from " << entry->start_ea << std::dec;
+
+      auto lp_entry = GetOrCreateBlock(ctx, entry->lp_ea);
+      ir.CreateBr(lp_entry);
+      ctx.lp_to_block[entry->lp_ea] = landing_bb;
+
+    } else {
+      std::stringstream ss;
+      ss << "catch_landingpad_" << std::hex << entry->lp_ea;
+      auto landing_bb = llvm::BasicBlock::Create(*gContext, ss.str(), ctx.lifted_func);
+      llvm::IRBuilder<> ir(landing_bb);
+      auto num_clauses = entry->type_var.size();
+      auto exn_type = llvm::StructType::get(llvm::Type::getInt8PtrTy(*gContext),
+                                            llvm::Type::getInt32Ty(*gContext), nullptr);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      auto lpad = ir.CreateLandingPad(exn_type, static_cast<unsigned int>(num_clauses), "catch.lpad");
+#else
+      auto personality = gModule->getFunction(PERSONALITY_FUNCTION);
+      auto lpad = ir.CreateLandingPad(exn_type, personality, static_cast<unsigned int>(num_clauses), "catch.lpad");
+#endif
+      auto catch_all = false;
+
+      // The type indices in exception table are reversed for some of the binaries. In such case, the
+      // wrong exception handler may get called.
+      // TODO(kumarak): Fix the wrong indices in the exception table.
+      for (auto type = entry->type_var.rbegin(); type != entry->type_var.rend(); type++) {
+        if ((*type)->ea) {
+          lpad->addClause(gModule->getGlobalVariable((*type)->name));
+        } else {
+          catch_all = true;
+        }
+      }
+
+      if (catch_all) {
+        lpad->addClause(llvm::Constant::getNullValue(ir.getInt8PtrTy()));
+      }
+
+      std::vector<llvm::Value *> args(2);
+#if LLVM_VERSION_NUMBER > LLVM_VERSION(3, 6)
+      args[0] = ir.CreateLoad(llvm::Type::getInt64Ty(*gContext), cfg_func->stack_ptr_var);
+      args[1] = ir.CreateLoad(llvm::Type::getInt64Ty(*gContext), cfg_func->frame_ptr_var);
+#else
+      args[0] = ir.CreateLoad(cfg_func->sp_var, true);
+      args[1] = ir.CreateLoad(cfg_func->bp_var, true);
+#endif
+      auto handler = GetExceptionHandlerPrologue();
+      ir.CreateCall(handler, args);
+      auto lp_entry = ctx.ea_to_block[entry->lp_ea];
+      if(lp_entry == nullptr) {
```

Comment:
Add a `LOG_IF` here like the one above to warn about there being a missing landing pad block, then handle it.

Ideally I'd like to see you refactor this function. Both sides of the `if` statement on the `entry->action_index` contain duplicated code.